File: Sample Apps/Objc/SampleApp/ConversationViewController.m
//
//  ConversationViewController.m
//  SampleApp
//
//  Created by Nimrod Shai on 2/23/16.
//  Copyright © 2016 LivePerson. All rights reserved.
//

#import "ConversationViewController.h"

@implementation ConversationViewController

- (IBAction)backButtonPressed:(id)sender {
    [[LPMessaging instance] removeConversation:self.conversationQuery];
    [[self navigationController] popViewControllerAnimated:true];
}

- (IBAction)menuButtonPressed:(id)sender {
    UIAlertControllerStyle style = UIAlertControllerStyleActionSheet;
    
    if ([[UIDevice currentDevice] userInterfaceIdiom] == UIUserInterfaceIdiomPad) {
        style = UIAlertControllerStyleAlert;
    }
    
    UIAlertController *alertController = [UIAlertController
                                          alertControllerWithTitle:@"Menu"
                                          message:@"Choose an option"
                                          preferredStyle:style];
    
    /**
     This is how to resolve a conversation
     */
    UIAlertAction *resolveAction = [UIAlertAction actionWithTitle:@"Resolve" style:UIAlertActionStyleDefault handler:^(UIAlertAction * _Nonnull action) {
        [[LPMessaging instance] resolveConversation:self.conversationQuery];
    }];
    
    
    NSString *urgentTitle = [[LPMessaging instance] isUrgent:self.conversationQuery] ? @"Dismiss Urgent" : @"Mark as Urgent";
    
    /**
     This is how to manage the urgency state of the conversation
     */
    UIAlertAction *urgentAction = [UIAlertAction actionWithTitle:urgentTitle style:UIAlertActionStyleDefault handler:^(UIAlertAction * _Nonnull action) {
        
        if ([[LPMessaging instance] isUrgent:self.conversationQuery]) {
            [[LPMessaging instance] dismissUrgent:self.conversationQuery];
        } else {
            [[LPMessaging instance] markAsUrgent:self.conversationQuery];
        }
    }];
    
    UIAlertAction *cancelAction = [UIAlertAction actionWithTitle:@"Cancel" style:UIAlertActionStyleCancel handler:^(UIAlertAction * _Nonnull action) {
        
    }];
    
    [alertController addAction:resolveAction];
    [alertController addAction:urgentAction];
    [alertController addAction:cancelAction];
    
    [self presentViewController:alertController animated:true completion:^{
        
    }];
}

@end

File: Sample Apps/Swift/SampleApp-Swift/ConversationViewController.swift
//
//  ConversationViewController.swift
//  SampleApp-Swift
//
//  Created by Nimrod Shai on 2/23/16.
//  Copyright © 2016 LivePerson. All rights reserved.
//

import Foundation
import LPMessagingSDK

class ConversationViewController: UIViewController {

    var account: String? = nil
    var conversationQuery: ConversationParamProtocol?
    
    @IBAction func backButtonPressed() {
        if self.account != nil {
            self.conversationQuery = LPMessaging.instance.getConversationBrandQuery(self.account!)
            if self.conversationQuery != nil {
                LPMessaging.instance.removeConversation(self.conversationQuery!)
                _ = self.navigationController?.popViewController(animated: true)
            }
        }
    }
    
    @IBAction func menuButtonPressed() {
        var style = UIAlertController.Style.actionSheet
        
        if UIDevice.current.userInterfaceIdiom == .pad {
            style = .alert
        }
        
        let alertController = UIAlertController(title: "Menu", message: "Choose an option", preferredStyle: style)
        
        /**
        is how to resolve a conversation
        */
        let resolveAction = UIAlertAction(title: "Resolve", style: .default) { (alert: UIAlertAction) -> Void in
            if self.conversationQuery != nil {
                LPMessaging.instance.resolveConversation(self.conversationQuery!)
            }
        }
        
        let urgentTitle = LPMessaging.instance.isUrgent(self.conversationQuery!) ? "Dismiss Urgent" : "Mark as Urgent"
     
        /**
        This is how to manage the urgency state of the conversation
        */
        let urgentAction = UIAlertAction(title: urgentTitle, style: .default) { (alert: UIAlertAction) -> Void in
            if LPMessaging.instance.isUrgent(self.conversationQuery!) {
                LPMessaging.instance.dismissUrgent(self.conversationQuery!)
            } else {
                LPMessaging.instance.markAsUrgent(self.conversationQuery!)
            }
        }
        
        let cancelAction = UIAlertAction(title: "Cancel", style: .cancel) { (alert: UIAlertAction) -> Void in
            
        }
        
        alertController.addAction(resolveAction)
        alertController.addAction(urgentAction)
        alertController.addAction(cancelAction)
        
        self.present(alertController, animated: true) { () -> Void in
            
        }
    }
}
